src/test/java/sg/dex/starfish/impl/operations/TestMemoryOperations.java

```diff
@@ -154,6 +176,10 @@ public void testHashAsyncFailed()  {
         } catch (Exception e) {
             /* OK, Expected */
         }
+        String provenance= epicFailOperation.getMetadata().get(Constant.PROVENANCE).toString();
```

Comment:
I think a Asset.getProvenance() method is appropriate as a shortcut for this. Will be a pretty important method.